dlt/extract/utils.py

```diff
@@ -183,6 +183,17 @@ def check_compat_transformer(name: str, f: AnyFun, sig: inspect.Signature) -> in
     return meta_arg
 
 
+def wrap_iterator(gen: Iterator[TDataItems]) -> Iterator[TDataItems]:
```

Comment:
pure iterators cannot be stopped, so we need to wrap them in a generator that I can control from the limit step.

Reply:
OK I have afterthought here. This is applied unconditionally to all `gen` objects? also to simple lists when there's no Limit step?

if so this will slow down simple pipelines considerably. maybe we should wrap in `bind` of `Limit`?